src/frontend/src/binder/expr/function.rs

```diff
@@ -201,16 +201,20 @@ impl Binder {
                     .unwrap_or_else(|_| ExprImpl::literal_null(DataType::Varchar)));
             }
             "current_schemas" => {
-                if inputs.len() != 1
-                    || (!inputs[0].is_null() && inputs[0].return_type() != DataType::Boolean)
-                {
-                    return Err(ErrorCode::ExprError(
+                let no_match_err = ErrorCode::ExprError(
                         "No function matches the given name and argument types. You might need to add explicit type casts.".into()
-                    )
-                    .into());
+                    );
+                if inputs.len() != 1 {
+                    return Err(no_match_err.into());
                 }
-
-                let ExprImpl::Literal(literal) = &inputs[0] else {
+                let input = inputs
+                    .pop()
+                    .unwrap()
+                    .enforce_bool_clause("current_schemas")
+                    .map_err(|_| no_match_err)?;
+
+                println!("input: {:?}", input);
+                let ExprImpl::Literal(literal) = &input else {
                     return Err(ErrorCode::NotImplemented(
```

Comment:
We may support this in some other PR. For now just fix implicit cast for `current_schemas`.

If this should be easy fix, feel free to suggest.

Reply:
This will be solved once we generalize another place from handling `null` to `unknown` types: #7320